MIPs/mip-mc-handshake.md

```diff
@@ -0,0 +1,255 @@
+Status: Draft
+Stability: n/a
+discussions-to: https://github.com/MetaMask/metamask-improvement-proposals/discussions
+Author(s): Vandan Parikh (@vandan), Alex Donesky (@adonesky1)
+Type: Maintainer
+Created: 2024-01-25
+---
+
+## Summary
+
+A proposal to introduce an unrestricted `provider_authorize` method for applications to connect to MetaMask per ChainAgnostic Standards Alliance’s CAIP-25 standard. The `provider_authorize` method establishes an interface for permissioning application access to wallet-managed resources including accounts, RPC providers, and specific JSON-RPC methods across multiple chains (EVM and beyond) in a concurrent session. An application can request access to these resources by specifying scopes as defined in CAIP-217.
+
+This interface is intended to be part of a new version of the MetaMask Wallet API that is optimized for multichain interactions.
+
+This proposal aims to implement CAIP-25's JSON-RPC Provider Authorization into MetaMask. For EVM (eip155) networks, the `rpcEndpoints` parameter is adapted to be compliant with the object structure defined in EIP-3085. Designing the interface in this way will provide a standardized approach for dApps to request access to networks and JSON-RPC methods.
+
+## Motivation
+
+The purpose of this proposal is to offer a convenient and standard interface for performing an initial handshake with a version of MetaMask that is designed for simultaneous interactions across multiple chains. 
+
+The provider_authorize method is specifically meant to:
+
+- Generally improve application-to-wallet interface negotiation
+- Initiate multichain account and network connections
+- Precisely negotiate an initial set of permissions with MetaMask
+
+By adopting the CAIP-25 standard from the ChainAgnostic Standards Alliance to accomplish this, MetaMask can encourage industry interoperability while building a unique and powerful multichain experience.
+
+General motivations for introducing a multichain-optimized Wallet API include:
+
+- Reducing friction when interacting across different chains (both in dApps and MetaMask)
+- Simplifying development for dApp builders
+- Aligning MetaMask with community standards to simplify integrations and tooling in the ecosystem
+- Increasing MetaMask's utility as a multichain wallet
+
+This proposal is fundamental to evolving MetaMask to support a safe and intuitive experience when interacting with multichain dapps. Supporting a breadth of interactions across these chains, while retaining a logical developer and end-user experience, promises to unlock further ecosystem growth.
+
+## Usage Example
+
+### Request
+An example of a typical JSON-RPC request that an application would send to request authorization(s) from a provider:
+
+```js
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "provider_authorize",
+  "params": {
+    "optionalScopes":{ /* only emphasize optionalScopes, indicate that required scopes are possible elsewhere */
```

Comment:
should this comment be updated? reads like a TODO item

Reply:
yes, those are old notes, which I intend to update or remove.